src/BuildScriptGeneratorCli/Commands/BuildpackDetectCommand.cs

```diff
@@ -0,0 +1,98 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Linq;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator;
+using Microsoft.Oryx.Common;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command("buildpack-detect", Description = "Determines whether Oryx can be applied as a buildpack to " +
+        "an app in the current working directory.")]
+    internal class BuildpackDetectCommand : CommandBase
+    {
+        // CodeDetectFail @ https://github.com/buildpack/lifecycle/blob/master/detector.go
+        public const int DetectorFailCode = 100;
+
+        [Argument(0, Description = "The source directory.")]
+        public string SourceDir { get; set; }
+
+        [Option("--platform-dir <dir>", CommandOptionType.SingleValue, Description = "Platform directory path.")]
+        public string PlatformDir { get; set; }
+
+        [Option("--plan-path <path>", CommandOptionType.SingleValue, Description = "Build plan TOML path.")]
+        public string PlanPath { get; set; }
+
+        internal override bool IsValidInput(IServiceProvider serviceProvider, IConsole console)
+        {
+            var result = true;
+            var options = serviceProvider.GetRequiredService<IOptions<BuildScriptGeneratorOptions>>().Value;
+            var logger = serviceProvider.GetService<ILogger<BuildpackDetectCommand>>();
+
+            if (!Directory.Exists(options.SourceDir))
+            {
+                logger.LogError("Could not find the source directory {srcDir}", options.SourceDir);
+                console.Error.WriteLine($"Error: Could not find the source directory '{options.SourceDir}'.");
+                result = false;
+            }
+
+            if (!File.Exists(PlanPath))
```

Comment:
Sanitize this and as well as `PlatformDir` too

Reply:
you mean with `Path.GetFullPath`?

Reply:
yes